tests/integration/Core/Image/ImageStorage/ImageStorageTest.php

```diff
@@ -0,0 +1,207 @@
+use eZ\Publish\Core\IO\IOServiceInterface;
+use eZ\Publish\Core\IO\MetadataHandler;
+use eZ\Publish\Core\IO\UrlRedecoratorInterface;
+use eZ\Publish\Core\IO\Values\BinaryFile;
+use eZ\Publish\Core\IO\Values\BinaryFileCreateStruct;
+use eZ\Publish\SPI\Persistence\Content\Field;
+use eZ\Publish\SPI\Persistence\Content\FieldValue;
+use eZ\Publish\SPI\Persistence\Content\VersionInfo;
+
+final class ImageStorageTest extends BaseCoreFieldTypeIntegrationTest
+{
+    /** @var \eZ\Publish\Core\FieldType\Image\ImageStorage\Gateway|\PHPUnit\Framework\MockObject\MockObject */
+    private $gateway;
+
+    /** @var \eZ\Publish\Core\IO\UrlRedecoratorInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $redecorator;
+
+    /** @var \eZ\Publish\Core\IO\MetadataHandler|\PHPUnit\Framework\MockObject\MockObject */
+    private $imageSizeMetadataHandler;
+
+    /** @var \eZ\Publish\Core\FieldType\Image\PathGenerator|\PHPUnit\Framework\MockObject\MockObject */
+    private $pathGenerator;
+
+    /** @var \eZ\Publish\Core\Base\Utils\DeprecationWarnerInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $deprecationWarner;
+
+    /** @var \eZ\Publish\Core\FieldType\Image\AliasCleanerInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $aliasCleaner;
+
+    /** @var \eZ\Publish\Core\IO\FilePathNormalizerInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $filePathNormalizer;
+
+    /** @var \eZ\Publish\Core\IO\IOServiceInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $ioService;
+
+    /** @var \eZ\Publish\Core\FieldType\Image\ImageStorage|\PHPUnit\Framework\MockObject\MockObject */
+    private $storage;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->redecorator = $this->createMock(UrlRedecoratorInterface::class);
+        $this->gateway = new DoctrineStorage($this->redecorator, $this->getDatabaseConnection());
+        $this->imageSizeMetadataHandler = $this->createMock(MetadataHandler::class);
+        $this->pathGenerator = $this->createMock(PathGenerator::class);
+        $this->deprecationWarner = $this->createMock(DeprecationWarnerInterface::class);
+        $this->aliasCleaner = $this->createMock(AliasCleanerInterface::class);
+        $this->filePathNormalizer = $this->createMock(FilePathNormalizerInterface::class);
+        $this->ioService = $this->createMock(IOServiceInterface::class);
+        $this->storage = $this->getMockBuilder(ImageStorage::class)
+            ->onlyMethods([])
+            ->setConstructorArgs(
+                [
+                    $this->gateway,
+                    $this->ioService,
+                    $this->pathGenerator,
+                    $this->imageSizeMetadataHandler,
+                    $this->deprecationWarner,
+                    $this->aliasCleaner,
+                    $this->filePathNormalizer,
+                ]
+            )
+            ->getMock();
```

Comment:
Rule `#1` of writing unit tests: do not mock a class under test :-)

This should work fine:
```php
new ImageStorage(
    $this->gateway,
    $this->ioService,
    $this->pathGenerator,
    $this->imageSizeMetadataHandler,
    $this->deprecationWarner,
    $this->aliasCleaner,
    $this->filePathNormalizer,
);
```